app/assets/javascripts/application.js

```diff
@@ -136,3 +136,52 @@ $(document).ready(function(){
   });
 
 });
+
+$.rails.allowAction = function(link){
```

Comment:
The whole changes in application.js are well isolated. I would vote to move them all to other file like `styled_confirmation.js`